app/src/main/java/com/gh4a/utils/HtmlUtils.java

```diff
@@ -137,33 +137,31 @@ public void drawBackground(Canvas c, Paint p, int left, int right, int top, int
 
     /**
      * Rewrite relative URLs in HTML fetched e.g. from markdown files.
-     *
-     * @param html
-     * @param repoUser
-     * @param repoName
-     * @param branch
-     * @return
      */
     public static String rewriteRelativeUrls(final String html, final String repoUser,
             final String repoName, final String branch) {
-        final String baseUrl = "https://raw.github.com/" + repoUser + "/" + repoName + "/" + branch;
-        final StringBuffer sb = new StringBuffer();
-        final Pattern p = Pattern.compile("(href|src)=\"(\\S+)\"");
-        final Matcher m = p.matcher(html);
+        final String baseUrl = "https://github.com/" + repoUser + "/" + repoName + "/blob/" + branch;
+        String rewrittenHtml = rewriteUrlsInAttribute("href", html, baseUrl);
 
-        while (m.find()) {
-            String url = m.group(2);
+        final String baseUrlForImages = "https://raw.github.com/" + repoUser + "/" + repoName + "/" + branch;
+        return rewriteUrlsInAttribute("src", rewrittenHtml, baseUrlForImages);
+    }
+
+    private static String rewriteUrlsInAttribute(String attribute, String html, String baseUrl) {
+        final Matcher matcher = Pattern.compile("(" + attribute + ")=\"(\\S+)\"").matcher(html);
+        final StringBuffer sb = new StringBuffer();
```

Comment:
Since we have two iterations per rewrite now instead of one, I wonder whether it wouldn't make sense to check whether there's any match at all first, and if there isn't, return `html` right away?

Reply:
Then you would have 3 iterations in case you have a match and still one in any case... I don't think there is a concrete performance penalty that justifies a preliminary check, which would still involve a full text scan.

Reply:
Then lazy init `sb` in the first 'found match' run and return `html` instead of `sb` if it's still null.

Reply:
> Then lazy init sb in the first 'found match' run and return html instead of sb if it's still null.

I didn't fully understand what you meant with this, but it seemed to me that the goal was to avoid the copy of the `html` string in the StringBuffer.
I did another thing instead: I've set the initial capacity of the StringBuffer to the length of `html`, to minimize the number of copies/reallocations when appending to the buffer.
I think that any other change that would make the logic more complex is simply not worth it. In the best case you would only save a few memory allocations.